hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/S3Acl.java

```diff
@@ -0,0 +1,390 @@
+    }
+    BitSet acls = new BitSet(IAccessAuthorizer.ACLType.getNoOfAcls());
+    switch (permissionType) {
+    case FULL_CONTROL:
+      acls.set(IAccessAuthorizer.ACLType.ALL.ordinal());
+      break;
+    case WRITE_ACP:
+      acls.set(IAccessAuthorizer.ACLType.WRITE_ACL.ordinal());
+      break;
+    case READ_ACP:
+      acls.set(IAccessAuthorizer.ACLType.READ_ACL.ordinal());
+      break;
+    case WRITE:
+      acls.set(IAccessAuthorizer.ACLType.WRITE.ordinal());
+      acls.set(IAccessAuthorizer.ACLType.DELETE.ordinal());
+      acls.set(IAccessAuthorizer.ACLType.CREATE.ordinal());
+      break;
+    case READ:
+      acls.set(IAccessAuthorizer.ACLType.READ.ordinal());
+      acls.set(IAccessAuthorizer.ACLType.LIST.ordinal());
+      break;
+    default:
+      LOG.error("Failed to recognize S3 permission {}", permission);
+      throw S3ErrorTable.newError(INVALID_ARGUMENT, permission);
+    }
+    return acls;
+  }
+
+  public static List<OzoneAcl> s3AclToOzoneNativeAclOnVolume(
+      S3BucketAcl bucketAcl) throws OS3Exception {
+    List<OzoneAcl> ozoneAclList = new ArrayList<>();
+    List<Grant> grantList = bucketAcl.getAclList().getGrantList();
+    for (Grant grant : grantList) {
+      //  Only "CanonicalUser" is supported, which maps to Ozone "USER"
+      ACLIdentityType identityType = ACLIdentityType.getTypeFromGranteeType(
+          grant.getGrantee().getXsiType());
+      if (identityType != null && identityType.isSupported()) {
+        String permission = grant.getPermission();
+        BitSet acls = getOzoneAclOnVolumeFromS3Permission(permission);
+        OzoneAcl defaultOzoneAcl = new OzoneAcl(
+            IAccessAuthorizer.ACLIdentityType.USER,
+            grant.getGrantee().getId(), acls,
+            OzoneAcl.AclScope.DEFAULT);
+        OzoneAcl accessOzoneAcl = new OzoneAcl(
+            IAccessAuthorizer.ACLIdentityType.USER,
+            grant.getGrantee().getId(), acls,
+            OzoneAcl.AclScope.ACCESS);
+        ozoneAclList.add(defaultOzoneAcl);
+        ozoneAclList.add(accessOzoneAcl);
+      } else {
+        LOG.error("Grantee type {} is not supported",
+            grant.getGrantee().getXsiType());
+        throw S3ErrorTable.newError(NOT_IMPLEMENTED,
+            grant.getGrantee().getXsiType());
+      }
+    }
+    return ozoneAclList;
+  }
+
+  public static BitSet getOzoneAclOnVolumeFromS3Permission(String permission)
+      throws OS3Exception {
+    BitSet acls = new BitSet(IAccessAuthorizer.ACLType.getNoOfAcls());
+    ACLType permissionType = ACLType.getType(permission);
+    if (permissionType == null) {
+      throw S3ErrorTable.newError(S3ErrorTable.INVALID_ARGUMENT, permission);
+    }
+    switch (permissionType) {
+    case FULL_CONTROL:
+      acls.set(IAccessAuthorizer.ACLType.ALL.ordinal());
+      break;
+    case WRITE_ACP:
+      acls.set(IAccessAuthorizer.ACLType.WRITE.ordinal());
+      break;
+    case READ_ACP:
```

Comment:
In the design doc, the bucket read acl is mapped to read/list on both volume and bucket. The code mapped to read only on the volume. Can you confirm if the doc needs to be changed? https://docs.google.com/document/d/18loKS6qbMKh8kKkpPvma9X-fNIcxBVAcHulAQxZpI1c/edit#